lib/active_merchant/billing/gateways/moneris.rb

```diff
@@ -314,7 +332,15 @@ def authorization_from(response = {})
       end
 
       # Tests for a successful response from Moneris' servers
-      def successful?(response)
+      def successful?(action, response, threed_ds_transaction = false)
+        # See 9.4 CAVV Result Codes in https://developer.moneris.com/livedemo/3ds2/reference/guide/php
+        cavv_accepted = if threed_ds_transaction
+                          response[:cavv_result_code] && response[:cavv_result_code] == "2"
+                        else
+                          true
```

Comment:
Why is the default `true`?

Reply:
thanks for the review, Amit! it's true so that for NON 3DS transactions we accept whatever cavv condition there is (`nil` or not even present). For 3DS transactions we want to verify that it's `2`.

Will fix rubocop!